vassal-app/src/main/java/VASSAL/counters/Mat.java

```diff
@@ -234,6 +251,23 @@ public Rectangle boundingBox() {
     return piece.boundingBox();
   }
 
+  /**
+   * Draw the cargo. Used to generate a DragShadow for a Mat correctly showing the loaded cargo
+   */
+  public void drawCargo(Graphics g, int x, int y, Component obs, double zoom) {
```

Comment:
It worked for you in Almoravid because your layers where correct, but in my case, the cargo could be drawn under the mat and it was very offputting. This just draws the cargo a second time on top of the mat,